cli/bus.h

```diff
@@ -0,0 +1,158 @@
+#pragma once
+
+#include "helper.h"
+#include "influxdb_format.h"
+
+namespace bus
+{
+
+using bus_request_info = std::tuple<std::string, uint64_t, uint64_t>;
+
+inline std::vector<bus_request_info> get_bus_requests(common::sdp::DataPlaneInSharedMemory& sdp_data)
+{
+	uint64_t* requests = common::sdp::ShiftBuffer<uint64_t*>(sdp_data.dataplane_data, sdp_data.metadata_bus.start_bus_requests);
+	uint64_t* durations = common::sdp::ShiftBuffer<uint64_t*>(sdp_data.dataplane_data, sdp_data.metadata_bus.start_bus_durations);
```

Comment:
Use `auto` when initializing with a template cast to avoid duplicating the type name:
```cpp
auto* requests = common::sdp::ShiftBuffer<uint64_t*>(sdp_data.dataplane_data, sdp_data.metadata_bus.start_bus_requests);
auto* durations = common::sdp::ShiftBuffer<uint64_t*>(sdp_data.dataplane_data, sdp_data.metadata_bus.start_bus_durations);
```